File: src/com/mycompany/myapp/entities/Offre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

import java.util.Date;

/**
 *
 * @author asus
 */
public class Offre {
    private int id;
    private String titre;
    private Date date;
    private String description;
    private int id_coach;
    private int id_categorie;
    private int idCompte;

    public Offre(int id, String titre, Date date, String description, int id_coach, int id_categorie, int idCompte) {
        this.id = id;
        this.titre = titre;
        this.date = date;
        this.description = description;
        this.id_coach = id_coach;
        this.id_categorie = id_categorie;
        this.idCompte = idCompte;
    }

    public Offre() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getId_coach() {
        return id_coach;
    }

    public void setId_coach(int id_coach) {
        this.id_coach = id_coach;
    }

    public int getId_categorie() {
        return id_categorie;
    }

    public void setId_categorie(int id_categorie)
    {
        this.id_categorie = id_categorie;
    }

    public int getIdCompte() {
        return idCompte;
    }

    public void setIdCompte(int idCompte) {
        this.idCompte = idCompte;
    }

    @Override
    public String toString() {
        return "Offre{" + "id=" + id + ", titre=" + titre + ", date=" + date + ", description=" + description + ", id_coach=" + id_coach + ", id_categorie=" + id_categorie + ", idCompte=" + idCompte + '}';
    }

    public Offre(int id, String titre, Date date, String description) {
        this.id = id;
        this.titre = titre;
        this.date = date;
        this.description = description;
    }

    public Offre(String titre, Date date, String description) {
        this.titre = titre;
        this.date = date;
        this.description = description;
    }

    public Offre(String titre, Date date, String description, int id_categorie) {
        this.titre = titre;
        this.date = date;
        this.description = description;
        this.id_categorie = id_categorie;
    }

    public Offre(String titre, String description, Date date, int id_categorie, int idCompte) {

        this.titre = titre;
        this.description = description;
        this.date = date;
        this.id_categorie = id_categorie;
        this.idCompte = idCompte;

    }

}

File: src/Entity/Offre.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * Offre
 *
 * @ORM\Table(name="offre", indexes={@ORM\Index(name="test", columns={"id_categorie"}), @ORM\Index(name="test2", columns={"id_coach"})})
 * @ORM\Entity
 * @ORM\Entity(repositoryClass=OffreRepository::class)

 */
class Offre
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups("post:read")

     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="titre", type="string", length=255, nullable=false)
     *  @Assert\NotBlank(message="veuillez saisir votre titre")
     *  @Groups("post:read")

     */
    private $titre;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date", type="date", nullable=false)
     *  @Assert\NotBlank(message="veuillez saisir votre date")
     *  @Groups("post:read")

     */
    private $date;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=255, nullable=false)
     *  @Assert\NotBlank(message="Description is required")
     * @Assert\Length(
     * min = 2,
     * max = 50,
     * minMessage = "Your description must be at least {{ limit }} characters long",
     * maxMessage = "Your description cannot be longer than {{ limit }} characters"
     * )
     * @Groups("post:read")

     */
    private $description;

    /**
     * @var int|null
     *
     * @ORM\Column(name="idCompte", type="integer", nullable=true)
     * @Groups("post:read")

     */
    private $idCompte;

    /**
     * @var \Coach
     *
     * @ORM\ManyToOne(targetEntity="Coach")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_coach", referencedColumnName="id")
     * })
     * @Groups("post:read")

     */
    private $idCoach;

    /**
     * @var \Categoriesport
     *
     * @ORM\ManyToOne(targetEntity="Categoriesport")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_categorie", referencedColumnName="id")
     * })
     * @Groups("post:read")

     */
    private $idCategorie;

    public function getId(): ?int
    {
        return $this->id;
    }

    /**
     * @return int
     */
    public function getIdCompte()
    {
        return $this->idCompte;
    }

    /**
     * @param int $idCompte
     */
    public function setIdCompte($idCompte)
    {
        $this->idCompte = $idCompte;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getIdCoach(): ?Coach
    {
        return $this->idCoach;
    }

    public function setIdCoach(?Coach $idCoach): self
    {
        $this->idCoach = $idCoach;

        return $this;
    }

    public function getIdCategorie(): ?Categoriesport
    {
        return $this->idCategorie;
    }

    public function setIdCategorie(?Categoriesport $idCategorie): self
    {
        $this->idCategorie = $idCategorie;

        return $this;
    }


}
